### PR TITLE
Make produce and consumer use LUA script

### PR DESCRIPTION
**Pull request by kcudnik**
+ update compiler warnings and fix unittest

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -33,7 +33,50 @@ AC_ARG_ENABLE(gtest,
...
+CFLAGS_COMMON+=" -fPIC"
+CFLAGS_COMMON+=" -std=c++11"
+CFLAGS_COMMON+=" -Wall"
+CFLAGS_COMMON+=" -Wcast-align"
+CFLAGS_COMMON+=" -Wcast-qual"
+CFLAGS_COMMON+=" -Wconversion"
+CFLAGS_COMMON+=" -Wdisabled-optimization"
+CFLAGS_COMMON+=" -Werror"
+CFLAGS_COMMON+=" -Wextra"
+CFLAGS_COMMON+=" -Wextra"
```
Comment (kcudnik) [inline configure.ac]: fixed

File: common/consumertable.cpp [hunk 2]
```diff
@@ -51,77 +52,94 @@ ConsumerTable::~ConsumerTable()
...
+        "    table.insert(ret, k)\n"
+        "    table.insert(ret, v)\n"
+        "end\n"
+
+        "if op == 'get' or op == 'getresponse' then\n"
+        "return ret\n"
+        "end\n"
+
+        "local keyname = KEYS[4] .. ':' .. key\n"
+        "if key == '' then\n"
```
Comment (lguohan) [inline common/consumertable.cpp]: how can if key == '' happen?

Comment (kcudnik) [inline common/consumertable.cpp]: it can and its used actually :)

Comment (lguohan) [inline common/consumertable.cpp]: in what case, when the list is empty?

File: tests/redis_ut.cpp [hunk 3]
```diff
@@ -348,9 +362,8 @@ TEST(Table, test)
     vector<KeyOpFieldsValuesTuple> tuples;
     t.getTableContent(tuples);
 
-    unsigned int size_t = 2;
     cout << "Get total " << tuples.size() << " number of entries" << endl;
-    EXPECT_EQ(tuples.size(), size_t);
+    EXPECT_EQ(tuples.size(), 2);
```
Comment (stcheng) [inline tests/redis_ut.cpp]: To use unsigned int size_t = 2 to suppress the warning: 
../googletest/googletest/include/gtest/gtest.h:1392:11: error: comparison between signed and unsigned integer expressions [-Werror=sign-compare]
   if (lhs == rhs) {

Comment (stcheng) [inline tests/redis_ut.cpp]: btw, better not to use size_t name as size_t is a keyword in c++

File: common/consumertable.cpp (hunk 2 above)
Comment (lguohan) [inline common/consumertable.cpp]: when op is set, the key is not empty string.

**Pull request merged**